cloudinit/util.py

```diff
@@ -1736,6 +1736,13 @@ def hash_blob(blob, routine, mlen=None):
         return digest
 
 
+def hash_buffer(f: io.BufferedIOBase) -> bytes:
```

Comment:
Can we get a docstring for this? In general I prefer having docstrings for any new utility functions.

Is the purpose here over calling `hashlib.sha1(f.read()).digest()` so that we don't crash on very large files? I'm not criticizing, just making sure I understand the purpose.